src/components/sideBarContentEdit/SidebarContentEdit.tsx

```diff
@@ -38,7 +70,45 @@ const SidebarContentEdit = ({
   getUpdatedTranscript: () => TranscriptContent;
   saveTranscript: (updatedContent: TranscriptContent) => Promise<void>;
 }) => {
-  const { data: selectableListData, isLoading, error } = useGetMetaData();
+  const [path, setPath] = useState<string>("");
+  const [initialCount, setInitialCount] = useState<number>(1);
+  const { data: selectableListData } = useGetMetaData();
+  const { data: directoryPaths } = useGetRepoDirectories(path.slice(0, -1));
+  const [directoryList, setDirectoryList] = useState<IDir[] | []>([]);
+  useEffect(() => {
+    // we want the rootpath to load first before
+    if (directoryPaths && initialCount < 2) {
+      setPath(`${data.content.loc}/`);
+      setInitialCount(2);
+    }
+  }, [data.content.loc, directoryPaths, initialCount]);
+  useEffect(() => {
+    const convertDirStructure = (
```

Comment:
Does this create a new mutated instance ofthe directiory or it duplicates it?

Reply:
> Does this create a new mutated instance ofthe directiory or it duplicates it?

it creates a mutated instance.